infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -128,7 +129,9 @@
     private long maxConflictWildcard = Address.NOT_FOUND;
 
     private final long defaultCacheSize = Long.MAX_VALUE;
-    private final Cache<Integer, Long> conflictToGlobalTailCache;
+    private final long defaultPerStreamCacheSize = Long.MAX_VALUE;
+    private long perStreamCacheSize = defaultPerStreamCacheSize;
+    private final Cache<UUID, Cache<Integer, Long>> conflictToGlobalTailCache;
```

Comment:
I also think there is a general problem in this scheme (Not because of this code change). The object.hashCode() generated at the client side will not match for the same object from different clients unless they are forced to generate a consistent hashcode across all client jvm instances. This will lead to a lot of problems in a cluster scenario.